src/code/z_play.c

```diff
@@ -549,4 +2056,270 @@ void Play_FillScreen(GameState* thisx, s16 fillScreenOn, u8 red, u8 green, u8 bl
+        }
+
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_55_80) &&
+            ((scene == ENTR_SCENE_GREAT_BAY_COAST) || (scene == ENTR_SCENE_ZORA_CAPE))) {
+            gSaveContext.nextCutsceneIndex = 0xFFF0;
+        }
+
+        // "First cycle" Termina Field
+        if (INV_CONTENT(ITEM_OCARINA) != ITEM_OCARINA) {
+            if ((scene == ENTR_SCENE_TERMINA_FIELD) &&
+                (((void)0, gSaveContext.save.entrance) != ENTRANCE(TERMINA_FIELD, 10))) {
+                gSaveContext.nextCutsceneIndex = 0xFFF4;
+            }
+        }
+        //! FAKE:
+        gSaveContext.save.entrance =
+            Entrance_Create(((void)0, scene), spawn, ((void)0, gSaveContext.save.entrance) & 0xF);
+    }
+
+    GameState_Realloc(&this->state, 0);
+    KaleidoManager_Init(this);
+    ShrinkWindow_Init();
+    View_Init(&this->view, gfxCtx);
+    func_801A3EC0(0);
+    Quake_Init();
+    Distortion_Init(this);
+
+    for (i = 0; i < ARRAY_COUNT(this->cameraPtrs); i++) {
+        this->cameraPtrs[i] = NULL;
+    }
+
+    Camera_Init(&this->mainCamera, &this->view, &this->colCtx, this);
+    Camera_ChangeStatus(&this->mainCamera, CAM_STATUS_ACTIVE);
+
+    for (i = 0; i < ARRAY_COUNT(this->subCameras); i++) {
+        Camera_Init(&this->subCameras[i], &this->view, &this->colCtx, this);
+        Camera_ChangeStatus(&this->subCameras[i], CAM_STATUS_INACTIVE);
+    }
+
+    this->cameraPtrs[CAM_ID_MAIN] = &this->mainCamera;
+    this->cameraPtrs[CAM_ID_MAIN]->uid = CAM_ID_MAIN;
+    this->activeCamId = CAM_ID_MAIN;
+
+    func_800DFF18(&this->mainCamera, 0x7F);
+    Sram_Alloc(&this->state, &this->sramCtx);
+    Regs_InitData(this);
+    Message_Init(this);
+    GameOver_Init(this);
+    SoundSource_InitAll(this);
+    EffFootmark_Init(this);
+    Effect_Init(this);
+    EffectSS_Init(this, 100);
+    CollisionCheck_InitContext(this, &this->colChkCtx);
+    AnimationContext_Reset(&this->animationCtx);
+    Cutscene_Init(this, &this->csCtx);
+
+    if (gSaveContext.nextCutsceneIndex != 0xFFEF) {
+        gSaveContext.save.cutscene = gSaveContext.nextCutsceneIndex;
+        gSaveContext.nextCutsceneIndex = 0xFFEF;
+    }
+
+    if (gSaveContext.save.cutscene == 0xFFFD) {
+        gSaveContext.save.cutscene = 0;
+    }
+
+    if (gSaveContext.nextDayTime != 0xFFFF) {
+        gSaveContext.save.time = gSaveContext.nextDayTime;
+        gSaveContext.skyboxTime = gSaveContext.nextDayTime;
+    }
+
+    if ((gSaveContext.save.time >= CLOCK_TIME(18, 0)) || (gSaveContext.save.time < CLOCK_TIME(6, 30))) {
+        gSaveContext.save.isNight = true;
+    } else {
+        gSaveContext.save.isNight = false;
+    }
+
+    func_800EDDB0(this);
+
+    if (((gSaveContext.gameMode != 0) && (gSaveContext.gameMode != 1)) || (gSaveContext.save.cutscene >= 0xFFF0)) {
+        gSaveContext.unk_3DC0 = 0;
+        Magic_Reset(this);
+        gSaveContext.sceneLayer = (gSaveContext.save.cutscene & 0xF) + 1;
+        gSaveContext.save.cutscene = 0;
```

Comment:
```suggestion
        // The cutscene system carried over from OoT was originally designed to trigger automatically when `>= 0xFFF0`.
        // This hack turns off cutscene here so that the scripted cutscene isn't immediately played.
        // This allows the `CutsceneManager` system to take over to determine when to play the scripted cutscene
        // Not setting this to 0 here will cause issues as each scene layer now has a list of cutscene scripts, so it will be unresolved as to which cutscene script to play
        gSaveContext.save.cutscene = 0;
```
I'm not sure the best way to phrase this, I don't think it needs to be this verbose. But here's a first attempt to get the idea across of why this line is important

Reply:
```suggestion

        // Set `gSaveContext.save.cutscene` to 0 so that the `CutsceneManager` takes over, instead of automatically playing.
        gSaveContext.save.cutscene = 0;
```
Perhaps something like this?

Reply:
I think what you committed was fine. Thanks